sdk/template/test-resources.json

```diff
@@ -0,0 +1,81 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "baseName": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().name]",
+            "metadata": {
+                "description": "The base resource name."
+            }
+        },
+        "tenantId": {
+            "type": "string",
+            "defaultValue": "72f988bf-86f1-41af-91ab-2d7cd011db47",
+            "metadata": {
+                "description": "The tenant ID to which the application and resources belong."
+            }
+        },
+        "testApplicationOid": {
+            "type": "string",
+            "defaultValue": "b3653439-8136-4cd5-aac3-2a9460871ca6",
+            "metadata": {
+                "description": "The client OID to grant access to test resources."
+            }
+        },
+        "location": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]",
+            "metadata": {
+                "description": "The location of the resource. By default, this is the same as the resource group."
+            }
+        }
+    },
+    "variables": {
+        "azureKeyVaultUrl": "[format('https://{0}.vault.azure.net', parameters('baseName'))]"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.KeyVault/vaults",
+            "apiVersion": "2016-10-01",
+            "name": "[parameters('baseName')]",
+            "location": "[parameters('location')]",
+            "properties": {
+                "sku": {
+                    "family": "A",
+                    "name": "premium"
+                },
+                "tenantId": "[parameters('tenantId')]",
+                "accessPolicies": [
+                    {
+                        "tenantId": "[parameters('tenantId')]",
+                        "objectId": "[parameters('testApplicationOid')]",
+                        "permissions": {
+                            "secrets": [
+                                "get"
+                            ]
+                        }
+                    }
+                ]
+            }
+        },
+        {
+            "type": "Microsoft.KeyVault/vaults/secrets",
+            "name": "[concat(parameters('baseName'), '/TestSecret')]",
+            "apiVersion": "2016-10-01",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+              "[resourceId('Microsoft.KeyVault/vaults', parameters('baseName'))]"
+            ],
+            "properties": {
+              "value": "Very secret value"
+            }
+          }
+    ],
+    "outputs": {
+        "KEYVAULT_URL": {
```

Comment:
AZURE_KEYVAULT_URL

Reply:
Why?

Reply:
Because that's what the guidelines state, and what we're discussing in another bug.

Reply:
Guidelines are about client libraries not test infrastructure.

Reply:
We're asking contributors to set environment variables. These scripts aren't just for us. The point is to make PII easy to identify and clean up, rather than spread sprawl them across an environment block to the point someone may miss PII and secrets leak. That's concerning. Grouping them with a common prefix makes them easy to identify and remove.

Reply:
We should group them but do it in a common place in the New-TestEnvironment script not throughout tens of `test-resources.json` files

Reply:
I don't see how that has anything to do with naming them with a common prefix. Yes, a common prefix should be done consistently in all locations as you pointed out in the other issue.

Reply:
We control how these outputs are turned into environment variables in a `New-TestEnvironment` script. With one change there we can make all the environment variables it creates compliant with any rules we want and not require prefixes in templates at all.

Reply:
And BTW this follows the guidelines for test variables precisely: https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/2/Azure-DevOps-Pipeline-Guidance?anchor=environment-variables 

```
Environment variables should follow the convention of <service>_<variable>. For example, if you are running live tests and need an environment variable for an Event Hubs instance name, the name would be EVENTHUB_NAME. The parts are

Service: EVENTHUB
Variable: NAME
```